### DISCUSSION TITLE
Suitable replacement for DerApplicationSpecific

### DISCUSSION THREAD

**Discussion by Janez-Z** [bcgit/bc-csharp#530] [Q&A]
I am rewriting an older project that was using BouncyCastle 1.8.6 for (among other things) generating certificates with a completely diffrent structure to x509. This made heavy use of the "DerApplicationSpecific" class, which consists of an integer tag and byte array for its content.
This does not exist in the newer versions(i am using 2.2.1).

After some searching i have not found any suitable replacements. Is there anything i could use or any additional libraries that offer similar functionality?

**Comment by Janez-Z** [comment]
It turns out that DerTaggedObject Has enough configurability to work.
By setting its isExplicit and tagClass it works correctly.
If anyone has similar issues here is a pair of helper functions that do that and work as a drop in replacement: 

```
public static DerTaggedObject GetAppliationSpecific(int tagNumber, byte[] content)
{
       return new DerTaggedObject(false, 64, tagNumber, new DerOctetString(content));
}
public static DerTaggedObject GetAppliationSpecific(int tagNumber, Asn1Encodable content)
{
       return new DerTaggedObject(false, 64, tagNumber, content);
}
```

**Comment by Gonmore** [comment]
did you found any way to resolve this? I am having the same issue

**Reply by peterdettman** [reply]
Generally speaking, `DerApplicationSpecific` can be replaced by `DerTaggedObject` with tag class `Asn1Tags.Application` (0x40). When encoding, the DerTaggedObject should be created with that tag class, and when decoding an expected DerApplicationSpecific, the tag class should be validated (along with the tag number). Post code sample if you need more specific help.
